Metadata/MetadataConverter.php

```diff
@@ -64,6 +64,14 @@ public function convert(ClassHierarchyMetadata $metadata)
             $definition->setTags($classMetadata->tags);
             $definition->setProperties($classMetadata->properties);
 
+            if (null !== $classMetadata->decorates && method_exists($definition, 'setDecoratedService')) {
+                $definition->setDecoratedService($classMetadata->decorates, $classMetadata->decoration_inner_name);
```

Comment:
sry mixed 2 things in my head -> exception makes sense for the decorates annotation!

only for the deprecation warning I would not add an exeption